src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -92,25 +90,25 @@ struct SchedulingAlgorithmState {
   /** An array of pointers to workers in the worker pool. These are workers
    *  that have registered a PID with us and that are now waiting to be
    *  assigned a task to execute. */
-  UT_array *available_workers;
+  std::vector<LocalSchedulerClient *> available_workers;
   /** An array of pointers to workers that are currently executing a task,
    *  unblocked. These are the workers that are leasing some number of
    *  resources. */
-  UT_array *executing_workers;
+  std::vector<LocalSchedulerClient *> executing_workers;
   /** An array of pointers to workers that are currently executing a task,
    *  blocked on some object(s) that isn't available locally yet. These are the
    *  workers that are executing a task, but that have temporarily returned the
    *  task's required resources. */
-  UT_array *blocked_workers;
+  std::vector<LocalSchedulerClient *> blocked_workers;
   /** A hash map of the objects that are available in the local Plasma store.
    *  The key is the object ID. This information could be a little stale. */
-  object_entry *local_objects;
+  std::unordered_map<ObjectID, ObjectEntry *, UniqueIDHasher> local_objects;
   /** A hash map of the objects that are not available locally. These are
    *  currently being fetched by this local scheduler. The key is the object
    *  ID. Every LOCAL_SCHEDULER_FETCH_TIMEOUT_MILLISECONDS, a Plasma fetch
    *  request will be sent the object IDs in this table. Each entry also holds
    *  an array of queued tasks that are dependent on it. */
-  object_entry *remote_objects;
+  std::unordered_map<ObjectID, ObjectEntry *, UniqueIDHasher> remote_objects;
```

Comment:
This one and `local_objects` can just be `std::unordered_map<ObjectID, ObjectEntry, UniqueIDHasher>`, to avoid allocation/deallocation.